Google.Api.Gax.Grpc/Rest/RestMethod.cs

```diff
@@ -82,19 +78,124 @@ internal static RestMethod Create(MethodDescriptor method, JsonParser parser)
         /// <returns>A request with the URI, method and content populated.</returns>
         internal HttpRequestMessage CreateRequest(IMessage protoRequest, string host)
         {
-            string path = _pathPattern.Format(protoRequest);
-            var uri = host is null ? new Uri(path, UriKind.Relative) : new UriBuilder { Host = host, Path = path }.Uri;
-            string jsonContent = _contentFactory?.Invoke(protoRequest);
+            var transcodingResult = Transcode(protoRequest);
+
+            var uriPathWithParams = AppendQueryStringParameters(transcodingResult.UriPath, transcodingResult.QueryStringParameters.OrderBy(kvp => kvp.Key));
```

Comment:
Possible optimization: for the majority case where there aren't any query string parameters, skip the call.

Reply:
👍